src/connectionhealthpolicy/BaseConnectionHealthPolicy.ts

```diff
@@ -10,12 +10,22 @@ export default class BaseConnectionHealthPolicy implements ConnectionHealthPolic
   protected minHealth: number;
   protected maxHealth: number;
   protected currentHealth: number;
+  readonly canonicalNameForLogging?: string;
 
-  constructor(configuration: ConnectionHealthPolicyConfiguration, data: ConnectionHealthData) {
+  constructor(
+    configuration: ConnectionHealthPolicyConfiguration,
+    data: ConnectionHealthData,
+    canonicalNameForLogging?: string
+  ) {
     this.minHealth = configuration.minHealth;
     this.maxHealth = configuration.maxHealth;
     this.currentHealth = configuration.initialHealth;
     this.currentData = data.clone();
+    this.canonicalNameForLogging = canonicalNameForLogging;
+  }
+
+  getCanonicalNameForLogging(): string {
```

Comment:
can't we just make the member public instead of manually writing a getter? we can always do `get XXX` later.